dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/programrule/ProgramRuleAssignActionTest.java

```diff
@@ -0,0 +1,181 @@
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.programrule.ProgramRule;
+import org.hisp.dhis.programrule.ProgramRuleAction;
+import org.hisp.dhis.programrule.ProgramRuleActionService;
+import org.hisp.dhis.programrule.ProgramRuleActionType;
+import org.hisp.dhis.programrule.ProgramRuleService;
+import org.hisp.dhis.programrule.ProgramRuleVariable;
+import org.hisp.dhis.programrule.ProgramRuleVariableService;
+import org.hisp.dhis.setting.SettingKey;
+import org.hisp.dhis.setting.SystemSettingManager;
+import org.hisp.dhis.tracker.Assertions;
+import org.hisp.dhis.tracker.TrackerImportParams;
+import org.hisp.dhis.tracker.TrackerImportService;
+import org.hisp.dhis.tracker.TrackerImportStrategy;
+import org.hisp.dhis.tracker.TrackerTest;
+import org.hisp.dhis.tracker.report.ImportReport;
+import org.hisp.dhis.tracker.report.Warning;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class ProgramRuleAssignActionTest extends TrackerTest
+{
+    private static final String EVENT_UID = "D9PbzJY8bJO";
+
+    @Autowired
+    private TrackerImportService trackerImportService;
+
+    @Autowired
+    private ProgramRuleService programRuleService;
+
+    @Autowired
+    private ProgramRuleActionService programRuleActionService;
+
+    @Autowired
+    private ProgramRuleVariableService programRuleVariableService;
+
+    @Autowired
+    private SystemSettingManager systemSettingManager;
+
+    private Program program;
+
+    private DataElement dataElement1;
+
+    @Override
+    public void initTest()
+        throws IOException
+    {
+        ObjectBundle bundle = setUpMetadata( "tracker/simple_metadata.json" );
+        program = bundle.getPreheat().get( PreheatIdentifier.UID, Program.class, "BFcipDERJnf" );
+        dataElement1 = bundle.getPreheat().get( PreheatIdentifier.UID, DataElement.class, "DATAEL00001" );
+        DataElement dataElement2 = bundle.getPreheat().get( PreheatIdentifier.UID, DataElement.class, "DATAEL00002" );
+        ProgramRuleVariable programRuleVariable = createProgramRuleVariableWithDataElement( 'A', program,
+            dataElement2 );
+        programRuleVariableService.addProgramRuleVariable( programRuleVariable );
+
+        injectAdminUser();
+
+        assignProgramRule();
+        TrackerImportParams params = fromJson( "tracker/programrule/tei_enrollment_completed_event.json" );
+
+        trackerImportService.importTracker( params );
+    }
+
+    @Test
+    void shouldImportWithWarningWhenDataElementWithSameValueIsAssignedByAssignRule()
+        throws IOException
+    {
+        TrackerImportParams params = fromJson( "tracker/programrule/event_update_datavalue_same_value.json" );
+        params.setImportStrategy( TrackerImportStrategy.CREATE_AND_UPDATE );
+
+        ImportReport importReport = trackerImportService.importTracker( params );
+
+        Assertions.assertHasOnlyWarningsAndNoErrors( importReport.getValidationReport(),
```

Comment:
I think we have variants of the assertions where we just pass in the ImportReport. If it does not exist please create it as the other assertions have it.

So we don't have to do the `importReport.getValidationReport()` over and over. Please use that one in all tests 😄